tools/prow-auto-bumper/pullrequest.go

```diff
@@ -0,0 +1,161 @@
+		if oncaller != "" {
+			assignment = "/cc @" + oncaller
+		} else {
+			assignment = "Nobody is currently oncall, so falling back to Blunderbuss."
+		}
+	} else {
+		assignment = fmt.Sprintf("An error occurred while finding an assignee: `%s`.\nFalling back to Blunderbuss.", err)
+	}
+
+	return body + assignment
+}
+
+func getOncaller() (string, error) {
+	req, err := http.Get(oncallAddress)
+	if err != nil {
+		return "", err
+	}
+	defer req.Body.Close()
+	if req.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("HTTP error %d (%q) fetching current oncaller", req.StatusCode, req.Status)
+	}
+	oncall := struct {
+		Oncall struct {
+			ToolsInfra string `json:"tools-infra"`
+		} `json:"Oncall"`
+	}{}
+	if err := json.NewDecoder(req.Body).Decode(&oncall); err != nil {
+		return "", err
+	}
+	return oncall.Oncall.ToolsInfra, nil
+}
+
+func makeGitCommit(gi gitInfo, message string, dryrun bool) error {
+	if "" == gi.head {
+		log.Fatal("pushing to empty branch ref is not allowed")
+	}
+	if err := run(
+		"Running 'git add -A'",
+		func() error { return call("git", "add", "-A") },
+		dryrun); err != nil {
+		return fmt.Errorf("failed to git add: %v", err)
+	}
+	commitArgs := []string{"commit", "-m", message}
+	if "" != gi.userName && "" != gi.email {
+		commitArgs = append(commitArgs, "--author", fmt.Sprintf("%s <%s>", gi.userName, gi.email))
+	}
+	if err := run(
+		fmt.Sprintf("Running 'git %s'", strings.Join(commitArgs, " ")),
+		func() error { return call("git", commitArgs...) },
+		dryrun); err != nil {
+		return fmt.Errorf("failed to git commit: %v", err)
+	}
+	pushArgs := []string{"push", "-f", fmt.Sprintf("git@github.com:%s/%s.git", gi.userID, gi.repo),
+		fmt.Sprintf("HEAD:%s", gi.head)}
+	if err := run(
+		fmt.Sprintf("Running 'git %s'", strings.Join(pushArgs, " ")),
+		func() error { return call("git", pushArgs...) },
+		dryrun); err != nil {
+		return fmt.Errorf("failed to git push: %v", err)
+	}
+	return nil
+}
+
+// Get existing open PR not merged yet
+func getExistingPR(gcw *GHClientWrapper, gi gitInfo, matchTitle string) (*github.PullRequest, error) {
+	var res *github.PullRequest
+	PRs, err := gcw.ListPullRequests(gi.org, gi.repo, gi.getHeadRef(), gi.base)
+	if nil == err {
+		for _, PR := range PRs {
+			if string(ghutil.PullRequestOpenState) == *PR.State && strings.Contains(*PR.Title, matchTitle) {
+				res = PR
+				break
+			}
+		}
+	}
+	return res, err
+}
+
+func createOrUpdatePR(gcw *GHClientWrapper, pv *PRVersions, gi gitInfo, extraMsgs []string, dryrun bool) error {
+	title := makeCommitSummary(pv.getDominantVersions())
+	matchTitle := getMatchTitle()
+	body := generatePRBody(extraMsgs)
+	if err := makeGitCommit(gi, title, dryrun); nil != err {
+		return fmt.Errorf("failed git commit: '%v'", err)
+	}
+	existPR, err := getExistingPR(gcw, gi, matchTitle)
+	if nil != err {
+		return fmt.Errorf("failed querying existing pullrequests: '%v'", err)
+	}
+	if nil != existPR {
+		log.Printf("found open PR '%d'", *existPR.Number)
```

Comment:
Nit - some of log messages are capitalized but some are not. Better to make them consistent.

Reply:
Done.